refactor: extract babel setup helper into own module

buschtoens opened the pull request:
While working on https://github.com/ember-decorators/ember-decorators/pull/182#issuecomment-347255423 I hit a wall when trying to call this addon's `included` hook. This PR originally attempted to only call `this._super.included`, if it is set, which feels really hacky and needlessly convoluted.

Now it extracts the babel setup helper into its own module for easy consumption by other addons.

buschtoens committed:
fix(build): only call _super.included, if it is set

rwjblue reviewed (commented):
It seems really bizarre to need this, doesn’t it?

pzuraq commented:
I was thinking about this, a better solution may be to extract the function entirely into a separate module, and call it from both here and `ember-decorators`

buschtoens commented:
Sounds good. I'll do it next week. :)

buschtoens committed:
Revert "fix(build): only call _super.included, if it is set"

This reverts commit e9f65b696566658e0e6e891c2b099f9345a326ac.

buschtoens committed:
refactor: extract babel setup helper into own module

This allows us to explicitly call it from other addons, like in `ember-decorators`, where we need to do that for backwards compatibility reasons.

buschtoens commented on setup-babel.js:
```diff
@@ -0,0 +1,68 @@
+  // Create parent options, if they do nox exist
+  const parentOptions = (parent.options = parent.options || {});
+
+  const checker = new VersionChecker(parent).for('ember-cli-babel', 'npm');
```
This originally accessed `this.parent` instead of the `parent` argument that `included(parent)` receives. Can this change cause problems? I did not discover any so far.

buschtoens committed:
docs: fix typo 'nox' -> 'not'

buschtoens commented:
Should we document `setup-babel.js` as public API? I wouldn't since we added this only for backwards compatibility reasons. Addon authors should be encouraged to follow the [Installation instructions](https://github.com/ember-decorators/babel-transforms#installation) instead of fiddling around with these internals.

pzuraq commented:
I agree. We should probably actually explicitly document it as _not_ public API, either in the README or by putting it in some sort of `private` folder, so users can't complain if we change it.

We should still try not to change it until we make a major version revision, either way.

pzuraq reviewed (approved):
This looks great, I think this is definitely the right way to go until we reach 2.0

buschtoens committed:
refactor: move setup-babel to private directory

pzuraq commented:
Released in v0.1.1
